utils.py

```diff
@@ -0,0 +1,188 @@
+import pandas as pd
+import numpy as np
+import copy as copy
+import random
+
+
+def print_aggregations(aggregations):
```

Comment:
we should probably move this file into a folder rather than having it directly in the workspace... but you can do that another time, no need to block the PR for that!